TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -1544,48 +1544,61 @@ IJunctionRestrictionsManager junctionRestrictionsManager
                         }
 
                         if (sqrVelocity <= GlobalConfig.Instance.PriorityRules.MaxYieldVelocity
-                            * GlobalConfig.Instance.PriorityRules.MaxYieldVelocity) {
+                            * GlobalConfig.Instance.PriorityRules.MaxYieldVelocity ||
+                            Options.simulationAccuracy >= SimulationAccuracy.Medium) {
                             if (logPriority) {
                                 Log._Debug(
                                     $"VehicleBehaviorManager.MayChangeSegment({frontVehicleId}): " +
                                       $"{sign} sign. waittime={extVehicle.waitTime}");
                             }
 
-                            if (extVehicle.waitTime < GlobalConfig.Instance.PriorityRules.MaxPriorityWaitTime)
-                            {
+                            if (extVehicle.waitTime <
```

Comment:
From here and down, noticed the code has quite deep nesting. Just a warning sign that it is getting too complex.

Reply:
I know... That looks pretty good compared to previous implementation (before Harmony) :)